docker-selenium.yaml

```diff
@@ -173,42 +246,137 @@ subpackages:
+        - chromium-docker-selenium-compat
+        - chromium
+    pipeline:
       - working-directory: NodeChrome
         pipeline:
           - runs: |
-              cp wrap_chrome_binary ${{targets.subpkgdir}}/opt/bin/wrap_chrome_binary
-      # https://github.com/SeleniumHQ/docker-selenium/blob/trunk/Standalone/Dockerfile
-      - working-directory: Standalone
+              mkdir -p ${{targets.contextdir}}/opt/bin/
+              cp wrap_chrome_binary ${{targets.contextdir}}/opt/bin/
+    test:
+      environment:
+        environment:
+          SE_EVENT_BUS_PUBLISH_PORT: 4442
+          SE_EVENT_BUS_SUBSCRIBE_PORT: 4443
+          SE_EVENT_BUS_HOST: selenium-hub
+          CONFIG_FILE: /opt/selenium/config.toml
+          SE_BIND_HOST: true
+      pipeline:
+        - name: "Check selenium node-chrome status status"
+          uses: test/daemon-check-output
+          with:
+            start: "/opt/bin/start-selenium-node.sh"
+            timeout: 60
+            expected_output: |
+              Adding Chrome for
+              Started Selenium node
+
+  - name: ${{package.name}}-nodefirefox
+    dependencies:
+      runtime:
+        - ${{package.name}}-nodebase
+        - geckodriver
+        - firefox
+    pipeline:
+      - working-directory: NodeFirefox
         pipeline:
           - runs: |
-              cp start-selenium-standalone.sh ${{targets.subpkgdir}}/opt/bin/start-selenium-standalone.sh
-              cp selenium.conf ${{targets.subpkgdir}}/etc/supervisor/conf.d/
-              cp generate_config ${{targets.subpkgdir}}/opt/bin/generate_config
+              mkdir -p ${{targets.contextdir}}/opt/bin
+              mkdir -p ${{targets.contextdir}}/opt/selenium/
+              mkdir -p ${{targets.contextdir}}/etc/supervisor/conf.d/
+              cp firefox-cleanup.sh ${{targets.contextdir}}/opt/bin/firefox-cleanup.sh
+              cp firefox-cleanup.conf ${{targets.contextdir}}/etc/supervisor/conf.d/firefox-cleanup.conf
+              echo "firefox" > ${{targets.contextdir}}/opt/selenium/browser_name
+              firefox --version | awk '{print $3}' > ${{targets.contextdir}}/opt/selenium/browser_version
+              echo "\"moz:firefoxOptions\": {\"binary\": \"/usr/bin/firefox\"}" > ${{targets.contextdir}}/opt/selenium/browser_binary_location
+    test:
+      environment:
+        environment:
+          SE_EVENT_BUS_PUBLISH_PORT: 4442
+          SE_EVENT_BUS_SUBSCRIBE_PORT: 4443
+          SE_EVENT_BUS_HOST: selenium-hub
+          CONFIG_FILE: /opt/selenium/config.toml
+          SE_BIND_HOST: true
+      pipeline:
+        - runs: |
+            firefox --version
+        - name: "Check selenium node-firefox status status"
+          uses: test/daemon-check-output
+          with:
+            start: "/opt/bin/start-selenium-node.sh"
+            timeout: 60
+            expected_output: |
+              Adding Firefox for
+              Started Selenium node
+
+  - name: ${{package.name}}-standalone-chrome
```

Comment:
I think we can drop the meta package but otherwise, this looks great!

Reply:
it was agreed to keep this in for clarity